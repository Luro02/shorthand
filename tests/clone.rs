use shorthand::ShortHand;

#[derive(ShortHand, Default)]
struct Example {
    #[shorthand(enable(clone))]
    value: String,
}

#[test]
fn test_clone() {
    assert_eq!("".to_string(), Example::default().value());
}

fn main() {}
