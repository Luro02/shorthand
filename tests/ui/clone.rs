use shorthand::ShortHand;

struct NotClone;

#[derive(ShortHand)]
struct Example {
    #[shorthand(enable(clone))]
    value: NotClone,
}

fn main() {}
