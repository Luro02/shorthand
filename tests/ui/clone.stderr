error[E0277]: the trait bound `NotClone: std::clone::Clone` is not satisfied
 --> $DIR/clone.rs:8:12
  |
8 |     value: NotClone,
  |            ^^^^^^^^ the trait `std::clone::Clone` is not implemented for `NotClone`
  |
  = help: see issue #48214
  = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
